drivers/memc/Kconfig.stm32

```
@@ -17,3 +17,10 @@ config MEMC_STM32_SDRAM
 	select USE_STM32_HAL_SDRAM
 	help
 	  Enable STM32 FMC SDRAM controller.
+
+config MEMC_STM32_BACKUP_SRAM
+	bool "Enable STM32 Backup SRAM"
+	depends on SOC_SERIES_STM32H7X
```

I'd prefer we link this directly with device tree thanks to:
```suggestion
        bool
	default default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_BACKUP_SRAM))
```
And we get rid of `MEMC_STM32_BACKUP_SRAM` symbol selection by the user.

---

If we do this, then all boards with

```
&backup_sram {
    status = "okay";
};
```

will have the backup SRAM enabled. I'm not sure if this is what we want.

---

Right, application control is still required.